main.go

```diff
@@ -167,21 +170,36 @@ func main() {
 //   3. Run data conversion
 //   4. Generate report
 func toSpanner(driver, projectID, instanceID, dbName string, ioHelper *ioStreams, outputFilePrefix string, now time.Time) error {
-	conv, err := schemaConv(driver, ioHelper)
-	if err != nil {
-		return err
-	}
-	// close the seekable file
+	conv := internal.MakeConv()
+	var err error
+	if !dataOnly {
+		conv, err = schemaConv(driver, ioHelper)
+		if err != nil {
+			return err
+		}
+		// close the seekable file
 
-	if ioHelper.seekableIn != nil {
-		defer ioHelper.in.Close()
-	}
+		if ioHelper.seekableIn != nil {
+			defer ioHelper.in.Close()
+		}
 
-	writeSchemaFile(conv, now, outputFilePrefix+schemaFile, ioHelper.out)
-	writeSessionFile(conv, now, outputFilePrefix+sessionFile, ioHelper.out)
-	if schemaOnly {
-		report(driver, nil, ioHelper.bytesRead, "", conv, outputFilePrefix+reportFile, ioHelper.out)
-		return nil
+		writeSchemaFile(conv, now, outputFilePrefix+schemaFile, ioHelper.out)
+		writeSessionFile(conv, now, outputFilePrefix+sessionFile, ioHelper.out)
+		if schemaOnly {
```

Comment:
The current code leads to the following confusing behaviour:
If I set both schemaOnly and dataOnly, then we will just get data conversion (i.e. same behaviour as dataOnly).

I think it would be better to simply exit if both flags are set and give an error.

Code-wise, I'm wondering if we should do some flag validation at the top of toSpanner, and check for consistency of dataOnly, schemaOnly and session.
i.e. have one block of code that checks for consistency and generates the right errors.

Reply:
We are using dataOnly and schemaOnly as modes (main.go file) as well as we are using them while conversion (conv.go file). Hence, I think processSchema and processData would add complexity in understanding. Please review the checks that I have added before the call to toSpanner():
https://github.com/cloudspannerecosystem/harbourbridge/blob/20a5a88a6ce89b030ddd0a024800f3eb1a174526/main.go#L158-L167

Reply:
Change looks good -- it addresses my concerns.

I agree with you about the confusion of convert.go.
We should probably rename dataOnly and schemaOnly modes in that file.
Having flags dataOnly and schemaOnly and modes dataOnly and schemaOnly that mean slightly different things is really confusing!

But let's not solve the issue in this PR.